source/frontend/src/hooks/providers/useProductProvider.ts

```diff
@@ -0,0 +1,41 @@
+import { AxiosError, AxiosResponse } from 'axios';
+import { useApiProducts } from 'hooks/pims-api/useApiProducts';
+import { useApiRequestWrapper } from 'hooks/pims-api/useApiRequestWrapper';
+import { IApiError } from 'interfaces/IApiError';
+import { Api_AcquisitionFile } from 'models/api/AcquisitionFile';
+import { useCallback, useMemo } from 'react';
+import { toast } from 'react-toastify';
+
+/**
+ * hook that retrieves Product information.
+ */
+export const useProductProvider = () => {
+  const { getProductFile } = useApiProducts();
+
+  const { execute: retrieveProductFile, loading: retrieveProductFileLoading } =
+    useApiRequestWrapper<
+      (projectId: number) => Promise<AxiosResponse<Api_AcquisitionFile | null, any>>
+    >({
+      requestFunction: useCallback(
+        async (productId: number) => await getProductFile(productId),
+        [getProductFile],
+      ),
+      requestName: 'retrieveProductFile',
+      onSuccess: useCallback(() => toast.success('File for project retrieved'), []),
```

Comment:
I think this falls into the new category of a TMI toast. The BVT feedback we received was that they like getting feedback on save, but in cases such as load this can be redundant since the user can tell if their project file was retrieved just by looking at if there is data on the screen.

Reply:
Would be a good idea to introduce more robust logging if we are replacing the toasts everywhere

Reply:
I think all success/errors are already logged as long as you use the apiRequestWrapper (via redux).